app/Ptr/PtrControllerTest.php

```diff
@@ -345,4 +365,16 @@ private function deletePtrs(): void {
   protected function url($ptr = null) {
     return $ptr ? sprintf('%s/%d', $this->url, $ptr->id) : $this->url;
   }
+
+  private function mockDns($host, $ip, $times) {
+    $this->dns
```

Comment:
this is fine and you don't need to go back and change this, but for future reference, you don't really need to mock the DNS service here. You could just mock the validator instead, since that is what this class depends on. We already know the validator works as intended because we tested it, so it's perfectly fine to just make sure the validator gets the right input and this class handles its output correctly.